### PR TITLE
Implement Conditional Options feature

### PR DESCRIPTION
**Pull request by josefalanga**
Now the option dictionary has an `available` index with a boolean value, true by default, unless a conditional expression is used in the line.

I feel is a bit hacky because the boolean left in the stack can be there for other reasons unknown to me, but couldn't find the case. Looks like everything the vm puts in the stack, is discarded when the operation is finished, except this one case.

Having a `<<if>>` without `<<endif>>` elsewhere is illegal syntax and won't compile. So I guess the only way a boolean can be left dangling in the stack is with this Conditional Options.

### CODE REVIEW

**Comment by josefalanga** [comment]
I took a peek at what [YarnSpinner is doing](https://github.com/YarnSpinnerTool/YarnSpinner/blob/355836214c3e29cfff46f5ea3f8543166c0ab16c/YarnSpinner/VirtualMachine.cs#L836), and there should be a fourth operand indicating if the line had a condition evaluated, but our project only gets three operands. Can we deserialize/enable this operand from Protobuf, or this is as good as it gets?

Edit: Forget what I said, we have the fourth operator, I'm fixing it rigth now.

File: examples/yarn2/shortcuts.yarn [hunk 1]
```diff
@@ -4,7 +4,7 @@ This is a test of shortcut functionality.
 
 -> Option 1
     Option 1 selected.
--> Option 2
+-> Option 2 <<if 0 > 1>>
     Option 2 selected.
 -> Option 3
```
Comment (weaversam8) [inline examples/yarn2/shortcuts.yarn]: This PR looks great! Do you mind adding a test case with a conditional option that resolves to true?

Comment (weaversam8) [reply examples/yarn2/shortcuts.yarn]: FYI, I fixed some issues with failing tests, so you may want to pull before making further changes :)

**Comment by josefalanga** [comment]
added the case for condition resolving to true.

also: running automated tests, nice work!

**Review by weaversam8 — APPROVED** [review]

**Comment by weaversam8** [comment]
Nice work!

**Pull request merged**